core/include/scipp/core/element/arithmetic.h

```diff
@@ -22,8 +22,33 @@ constexpr auto add_inplace_types =
              std::tuple<float, int64_t>, std::tuple<float, int32_t>,
              std::tuple<int64_t, bool>>;
 
+constexpr auto add_inplace_nan_types =
+    arg_list<double, float, int64_t, int32_t, std::tuple<double, float>,
+             std::tuple<float, double>, std::tuple<int64_t, int32_t>,
+             std::tuple<int32_t, int64_t>, std::tuple<double, int64_t>,
+             std::tuple<double, int32_t>, std::tuple<float, int64_t>,
+             std::tuple<float, int32_t>, std::tuple<int64_t, bool>>;
```

Comment:
Can we avoid this and reuse `add_inplace_types`? I suppose the problem is the `isnan` does not exist for all those types? Maybe an `if constexpr` can help (maybe in a heper lambda)?
```cpp
if constexpr(std::is_floating_point<T>)
  return isnan(x);
else
  return false;
}
```

Reply:
Yes, that is exactly the problem - I didn't like having to define the sub list of types again.

I did consider a solution along the lines of what you propose, but it is also misleading without reading the implementation. It supposes that `add_inplace_nan_types` is going to support time point etc.